src/helpers/protocol.helpers.ts

```diff
@@ -0,0 +1,58 @@
+import { MAX_UINT_256 } from '@/const'
+import { IUseContract } from '@/composables'
+import { IL1Factory } from '@/types/contracts/L1Factory'
+import type { InfoDashboardType } from '@/types'
+
+const findMatchingProtocols = (
+  l1Pools: IL1Factory.PoolViewStructOutput[],
+  l2Pools: IL1Factory.PoolViewStructOutput[],
+) => {
+  const l1Pool = l1Pools.find(l1Pool =>
+    l2Pools.some(l2Pool => l1Pool.protocol === l2Pool.protocol),
+  )
+
+  if (!l1Pool) {
+    return [null, null]
+  }
+
+  const l2Pool = l2Pools.find(l2Pool => l2Pool.protocol === l1Pool.protocol)
+
+  return [l1Pool, l2Pool]
+}
```

Comment:
Why are you trying to find `l2Pool` if you already had matching pools on the previous step? You're doing unnecessary iterations. Also this:
```ts
 const l1Pool = l1Pools.find(l1Pool =>
    l2Pools.some(l2Pool => l1Pool.protocol === l2Pool.protocol),
  )
```
has **O(N^2) time complexity**. It can be reduced to **O(N)** as I already mentioned in one of the previous PR's. https://github.com/MorpheusAIs/DashBoard/pull/59#discussion_r1715300042

```ts
  const l2DeployedPoolMap = new Map(l2DeployedPools.map(pool => [pool.protocol, pool]));

  for (const l1Pool of l1DeployedPools) {
    const l2Pool = l2DeployedPoolMap.get(l1Pool.protocol);

    if (l2Pool) {
      return [l1Pool, l2Pool]
    }
  }
  
  return [null, null]
```

Reply:
done